api/v1alpha1/release_types.go

```diff
@@ -138,128 +114,128 @@ type Release struct {
-
-		if r.Status.DeploymentStartTime == nil {
-			r.Status.DeploymentStartTime = &metav1.Time{Time: time.Now()}
-		}
+func (r *Release) MarkDeploying(message string) {
+	if r.Status.Deployment.StartTime != nil {
+		return
 	}
+
+	r.Status.Deployment.StartTime = &metav1.Time{Time: time.Now()}
+
+	conditions.SetConditionWithMessage(&r.Status.Conditions, deployedConditionType, metav1.ConditionUnknown, DeployingReason, message)
 }
 
-// MarkFailed registers the completion time and changes the Succeeded condition to False with
-// the provided reason and message.
-func (r *Release) MarkFailed(reason ReleaseReason, message string) {
-	if r.IsDone() && r.Status.CompletionTime != nil {
+// MarkDeploymentFailed registers the completion time and changes the deployment condition to False with
+// the provided message.
+func (r *Release) MarkDeploymentFailed(message string) {
+	if r.HasDeploymentFinished() && r.Status.Deployment.CompletionTime != nil {
 		return
 	}
 
-	r.Status.CompletionTime = &metav1.Time{Time: time.Now()}
-	r.setStatusConditionWithMessage(releaseConditionType, metav1.ConditionFalse, reason, message)
+	r.Status.Deployment.CompletionTime = &metav1.Time{Time: time.Now()}
+	conditions.SetConditionWithMessage(&r.Status.Conditions, deployedConditionType, metav1.ConditionFalse, DeploymentFailedReason, message)
 
-	go metrics.RegisterCompletedRelease(reason.String(), r.Status.ReleaseStrategy, r.Status.Target,
-		r.Status.StartTime, r.Status.CompletionTime, false)
+	go metrics.RegisterDeployedRelease(DeploymentFailedReason.String(), r.Status.Target, string(metav1.ConditionFalse),
+		r.Status.Deployment.StartTime, r.Status.Deployment.CompletionTime)
 }
 
-// MarkInvalid changes the Succeeded condition to False with the provided reason and message.
-func (r *Release) MarkInvalid(reason ReleaseReason, message string) {
-	if r.IsDone() {
+// MarkReleaseFailed registers the completion time and changes the released condition to False with
+// the provided reason and message.
+func (r *Release) MarkReleaseFailed(reason conditions.ConditionReason, message string) {
+	if r.HasReleaseFinished() && r.Status.Release.CompletionTime != nil {
 		return
 	}
 
-	r.setStatusConditionWithMessage(releaseConditionType, metav1.ConditionFalse, reason, message)
+	r.Status.Release.CompletionTime = &metav1.Time{Time: time.Now()}
+	conditions.SetConditionWithMessage(&r.Status.Conditions, releasedConditionType, metav1.ConditionFalse, reason, message)
 
-	go metrics.RegisterInvalidRelease(reason.String())
+	go metrics.RegisterCompletedRelease(reason.String(), r.Status.Release.Strategy, r.Status.Target,
+		r.Status.Release.StartTime, r.Status.Release.CompletionTime, false)
 }
 
-// MarkRunning registers the start time and changes the Succeeded condition to Unknown.
-func (r *Release) MarkRunning() {
-	if r.HasStarted() && r.Status.StartTime != nil {
+// MarkReleaseInvalid changes the Succeeded condition to False with the provided reason and message.
```

Comment:
it is in the release package. why was `MarkInvalid` not a good enough name?

Reply:
Because it was very generic and it wasn't clear if it was referring to the Release or to the Deployment. That MarkInvalid could happen again in the PostActions maybe.